### PR TITLE
fix(executor): Do not make unneeded `get pod` when no sidecars

### PR DESCRIPTION
**Pull request by alexec**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -350,7 +350,7 @@ dist/kustomize:
 	dist/kustomize version
 
 # generates several installation files
-manifests/install.yaml: $(CRDS) dist/kustomize
+manifests/install.yaml: $(MANIFESTS) dist/kustomize
```
Comment (alexec) [inline Makefile:353]: fixes `make codegen` not making manifests when changed

File: Makefile [hunk 2]
```diff
@@ -386,10 +386,12 @@ install: $(MANIFESTS) $(E2E_MANIFESTS) dist/kustomize
 	kubectl config set-context --current --namespace=$(KUBE_NAMESPACE)
 	@echo "installing PROFILE=$(PROFILE) VERSION=$(VERSION), E2E_EXECUTOR=$(E2E_EXECUTOR)"
 	dist/kustomize build --load_restrictor=none test/e2e/manifests/$(PROFILE) | sed 's/argoproj\//$(IMAGE_NAMESPACE)\//' | sed 's/:latest/:$(VERSION)/' | sed 's/containerRuntimeExecutor: docker/containerRuntimeExecutor: $(E2E_EXECUTOR)/' | kubectl -n $(KUBE_NAMESPACE) apply --prune -l app.kubernetes.io/part-of=argo -f -
+ifeq ($(PROFILE),stress)
```
Comment (alexec) [inline Makefile:389]: we don't need these unless we're running stress and this makes `make start` faster and more robust

File: hack/test-examples.sh [hunk 3]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 set -eu -o pipefail
 
-kubectl delete wf -l workflows.argoproj.io/test
+./dist/argo delete -l workflows.argoproj.io/test
```
Comment (alexec) [inline hack/test-examples.sh:4]: this make `make test-examples` more robust

**Review by simster7 — APPROVED** [review]
Rubber-stamp

**Pull request merged**